### PR TITLE
support for ebs snapshots

### PR DESCRIPTION
**Pull request by jhedev**

### CODE REVIEW

File: cmd/snapshotter/main.go [hunk 1]
```diff
@@ -2,39 +2,77 @@ package main
...
 	"github.com/aws/aws-sdk-go/service/lightsail"
 	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
 
-	"github.com/grid-x/lightsail-auto-snapshot/pkg/snapshot"
+	snapec2 "github.com/grid-x/aws-auto-snapshot/pkg/snapshot/ec2"
+	snaplightsail "github.com/grid-x/aws-auto-snapshot/pkg/snapshot/lightsail"
 )
 
+type Snapshotter interface {
```
Comment (philippfranke) [inline cmd/snapshotter/main.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	defaultBackupTag    = "backup"
+	defaultRetentionTag = "retention"
+
+	defaultSnapshotSuffix = "auto-snapshot"
+	defaultDeleteAfterTag = "_DELETE_AFTER"
+
+	defaultRetentionDays = 7 // Default are 7 days retention
+	defaultDescription   = "auto snapshot created by grid-x/aws-auto-snapshot"
+)
+
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go:25]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	suffix         string // snapshot suffix
+	backupTag      string
+	retentionTag   string
+	deleteAfterTag string
+
+	logger log.FieldLogger
+}
+
+type Opt func(*SnapshotManager)
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+func WithRetentionTag(t string) Opt {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+	return func(m *SnapshotManager) {
+		m.retentionTag = t
+	}
+}
+
+func WithBackupTag(t string) Opt {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+	return func(m *SnapshotManager) {
+		m.backupTag = t
+	}
+}
+
+func WithSnapshotSuffix(suf string) Opt {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+	return func(m *SnapshotManager) {
+		m.suffix = suf
+	}
+}
+
+func WithDeleteAfterTag(tag string) Opt {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+	return func(m *SnapshotManager) {
+		m.deleteAfterTag = tag
+	}
+}
+
+func NewSnapshotManager(client *ec2.EC2, opts ...Opt) *SnapshotManager {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+		if resp.NextToken == nil {
+			break
+		}
+		token = resp.NextToken
+	}
+
+	return result, nil
+}
+
+func (smgr *SnapshotManager) Snapshot(ctx context.Context) error {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

File: pkg/snapshot/ec2/ec2.go [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+			},
+		); err != nil {
+			logger.Error(err)
+			continue
+		}
+	}
+	return nil
+}
+
+func (smgr *SnapshotManager) Prune(ctx context.Context) error {
```
Comment (philippfranke) [inline pkg/snapshot/ec2/ec2.go]: 👵: Golinter likes to see a comment here.

**Review by philippfranke — CHANGES_REQUESTED** [review]
All about comments...

**Comment by jhedev** [comment]
Added linting to the CI and fixed all problems @philippfranke

**Comment by jhedev** [comment]
ping @gq0 @philippfranke

File: glide.yaml [hunk 11]
```diff
@@ -8,3 +8,5 @@ import:
   - service/lightsail
 - package: github.com/sirupsen/logrus
   version: ~1.0.3
+- package: gopkg.in/alecthomas/kingpin.v2
```
Comment (gq0) [inline glide.yaml:11]: Oh, a new dependency.

**Review by gq0 — APPROVED** [review]
ad-hoc it looks good

**Review by philippfranke — APPROVED** [review]
👍  LGTM

**Pull request merged**